### PR TITLE
Allow passing `--libc` setting to NPM lifecycle hooks to enable cross-compilation for glibc/musl.

### PR DESCRIPTION
**Pull request by sfc-gh-ptabor**
{ the PR is on top of: https://github.com/aspect-build/rules_js/pull/1318 }

Tools like https://github.com/mapbox/node-pre-gyp#versioning interpret `target_libc` setting. 
As I'm working on building for Alpine (musl based) containers, I need support for cross-compilation towards prescribed variant of libc. 

### Type of change

- New feature or functionality (change which adds functionality)

### Test plan

- Covered by existing test cases
- New test cases added
- Manual testing; please provide instructions so we can reproduce:

### CODE REVIEW

File: npm/private/test/yaml_tests.bzl [hunk 1]
```diff
@@ -509,7 +509,7 @@ def _parse_errors(ctx):
 
     asserts.equals(env, None, parse("""\
 asdljfk
-            sdlkfdslkjf
+            gisdlkfdslkjf
```
Comment (alexeagle) [inline npm/private/test/yaml_tests.bzl:512]: is this edit load-bearing?

**Review by alexeagle — COMMENTED** [review]
thanks! Could you rebase? Will let @gregmagolan do the approve I think

File: npm/private/test/yaml_tests.bzl (hunk 1 above)
Comment (sfc-gh-ptabor) [reply npm/private/test/yaml_tests.bzl]: Yes. I'm sorry. I thought I have rollbacked it.

**Comment by sfc-gh-ptabor** [comment]
Rebased. Thank you.

**Comment by alexeagle** [comment]
ping @gregmagolan

**Review by gregmagolan — APPROVED** [review]
lgtm. Thanks for the PR!

**Comment by gregmagolan** [comment]
Need to `bazel run //npm/private/test:write_npm_translate_lock` to green it up

**Comment by sfc-gh-ptabor** [comment]
> Need to `bazel run //npm/private/test:write_npm_translate_lock` to green it up

Thank you for checking. 

Done.

**Pull request merged**